### PR TITLE
Improve PointSet class testing

### LINKED ISSUE #3333: Add tests for `vtk_points`
`pyvista.vtk_points` doesn't have unit tests.  #3311 adds some tests, but there is more functionality to test more fully, including `force_float`.

> I noticed that `vtk_points` doesn't have any direct tests.  I added relevant tests for the functionality being tested here for my own sanity.  I can open an issue to fully flesh out the tests.

_Originally posted by @MatthewFlamm in https://github.com/pyvista/pyvista/issues/3311#issuecomment-1248276490_

### PR DESCRIPTION
**Pull request by sgaist**
### Overview

This pull request will implement additional tests for the PointSet class

### CODE REVIEW

**Comment by MatthewFlamm** [comment]
Looks good so far.  The windows tests look like they might use different types.  The other failures are due to our CI being very flaky recently.

**Comment by MatthewFlamm** [comment]
If you intend to fully close #3333, it would be good to also directly test https://github.com/pyvista/pyvista/blob/66ef2df7d396b5f1c858ccb02652e4f93f036410/pyvista/utilities/helpers.py#L465

File: tests/test_pointset.py [hunk 1]
```diff
@@ -106,3 +106,13 @@ def test_filters_return_pointset(sphere):
...
+
+
+@pytest.mark.parametrize(
+    "force_float,expected_data_type", [(False, vtk.VTK_LONG_LONG), (True, vtk.VTK_FLOAT)]
+)
+def test_pointset_force_float(force_float, expected_data_type):
+    np_points = np.array([[1, 2, 3]])
+    pset = pyvista.PointSet(np_points, force_float=force_float)
+
+    assert pset.points.GetDataType() == expected_data_type
```
Comment (adeak) [inline tests/test_pointset.py:118]: Could we do these tests in numpy land, checking `pset.points.dtype`? The issue is probably that "int" numpy arrays are `int32` on (even 64-bit) windows, and `int64` on any sane 64-bit system. Using VTK types adds an additional layer of ambiguity. I'd recommend creating `np_points` with an explicit `dtype=np.int64)` and then checking for the same dtype in the `assert`. This should fix platform-dependence.

Comment (sgaist) [reply tests/test_pointset.py:118]: Good point, please see #3396 which is what this merge request should have been originally be about.

**Comment by adeak** [comment]
For the record you don't have to force push changes: we always squash commits on merge, so no matter how complex your history is, we'll end up with a single commit of yours per PR. In fact force pushes have caused issues in the past when a contributor didn't notice that a pyvista member made some commits on the PR, which got lost on the next force push. So I think we somewhat prefer to keep adding new commits rather than force pushing :)

**Comment by sgaist** [comment]
@adeak I am used to build (and rebuild) clean histories for branches to keep all relevant commits hence the force push but it's noted. I'll avoid them in the future.

File: tests/test_pointset.py [hunk 2]
```diff
@@ -106,3 +106,157 @@ def test_filters_return_pointset(sphere):
...
+    np_points = np.array([[1, 2, 3]], np.int64)
+    pset = pyvista.PointSet(np_points, force_float=force_float)
+
+    assert pset.points.dtype == expected_data_type
+
+
+def test_center_of_mass():
+    np_points = np.array([[0.0, 0.0, 1.0], [1.0, 0.0, 0.0]])
+    pset = pyvista.PointSet(np_points)
+    assert np.array_equal(pset.center_of_mass(), np.array([0.5, 0.0, 0.5]))
```
Comment (MatthewFlamm) [inline tests/test_pointset.py:122]: This is looking great so far!

This comment applies to most, or possibly all, usages.  We should use `np.allclose` unless we know that the arrays will be unchanged.  In this example there may be numerical errors due to numerical precision.

There are a few failing tests for Windows due to this.

File: tests/test_pointset.py [hunk 3]
```diff
@@ -106,3 +106,157 @@ def test_filters_return_pointset(sphere):
...
+        pset.points,
+        np.array(
+            [
+                [7.0, 2.0, 3.0],
+                [4.0, 5.0, 6.0],
+                [
+                    1.0,
+                    8.0,
+                    9.0,
+                ],
```
Comment (tkoyama010) [inline tests/test_pointset.py:159]: Is there any reason to change the layout of the last line?

File: tests/test_pointset.py [hunk 4]
```diff
@@ -106,3 +106,157 @@ def test_filters_return_pointset(sphere):
...
+        pset.points,
+        np.array(
+            [
+                [1.0, 8.0, 3.0],
+                [4.0, 5.0, 6.0],
+                [
+                    7.0,
+                    2.0,
+                    9.0,
+                ],
```
Comment (tkoyama010) [inline tests/test_pointset.py:180]: Same with https://github.com/pyvista/pyvista/pull/3395#discussion_r986943643.

File: tests/test_pointset.py [hunk 5]
```diff
@@ -106,3 +106,157 @@ def test_filters_return_pointset(sphere):
...
+        pset.points,
+        np.array(
+            [
+                [1.0, 2.0, 9.0],
+                [4.0, 5.0, 6.0],
+                [
+                    7.0,
+                    8.0,
+                    3.0,
+                ],
```
Comment (tkoyama010) [inline tests/test_pointset.py:201]: Same with https://github.com/pyvista/pyvista/pull/3395#discussion_r986943643.

File: tests/test_pointset.py [hunk 6]
```diff
@@ -106,3 +106,157 @@ def test_filters_return_pointset(sphere):
...
+        np.array(
+            [
+                [
+                    7.0,
+                    8.0,
+                    9.0,
+                ],
+                [4.0, 5.0, 6.0],
+                [1.0, 2.0, 3.0],
+            ],
```
Comment (tkoyama010) [inline tests/test_pointset.py:223]: Same with https://github.com/pyvista/pyvista/pull/3395#discussion_r986943643.

File: tests/test_pointset.py (hunk 2 above)
Comment (akaszynski) [reply tests/test_pointset.py:122]:
```suggestion
    assert np.allclose(pset.center_of_mass(), [0.5, 0.0, 0.5])
```

File: tests/test_pointset.py (hunk 4 above)
Comment (akaszynski) [inline tests/test_pointset.py:180]:
```suggestion
                [7.0, 2.0, 9.0],
```

File: tests/test_pointset.py (hunk 5 above)
Comment (akaszynski) [inline tests/test_pointset.py:201]:
```suggestion
                [7.0, 8.0, 3.0],
```

File: tests/test_pointset.py [hunk 7]
```diff
@@ -106,3 +106,157 @@ def test_filters_return_pointset(sphere):
...
+    pset.flip_normal([1.0, 1.0, 1.0], inplace=True)
+    assert np.array_equal(
+        pset.points,
+        np.array(
+            [
+                [
+                    7.0,
+                    8.0,
+                    9.0,
+                ],
```
Comment (akaszynski) [inline tests/test_pointset.py:220]:
```suggestion
                [7.0, 8.0, 9.0],
```

File: tests/test_pointset.py [hunk 8]
```diff
@@ -106,3 +106,157 @@ def test_filters_return_pointset(sphere):
...
+            dtype=np.float32,
+        ),
+    )
+
+
+def test_rotate_x():
+    np_points = np.array([1, 1, 1])
+    pset = pyvista.PointSet(np_points)
+    pset.rotate_x(45, inplace=True)
+    assert np.array_equal(
```
Comment (akaszynski) [inline tests/test_pointset.py:233]:
```suggestion
    assert np.allclose(
```

File: tests/test_pointset.py [hunk 9]
```diff
@@ -106,3 +106,157 @@ def test_filters_return_pointset(sphere):
...
+        pset.points, np.array([[1.0000000e00, 1.1102230e-16, 1.4142135e00]], dtype=np.float32)
+    )
+
+
+def test_rotate_y():
+    np_points = np.array([1, 1, 1])
+    pset = pyvista.PointSet(np_points)
+    pset.rotate_y(45, inplace=True)
+    assert np.array_equal(
```
Comment (akaszynski) [inline tests/test_pointset.py:242]:
```suggestion
    assert np.allclose(
```

File: tests/test_pointset.py [hunk 10]
```diff
@@ -106,3 +106,157 @@ def test_filters_return_pointset(sphere):
...
+        pset.points, np.array([[1.4142135e00, 1.0000000e00, 1.1102230e-16]], dtype=np.float32)
+    )
+
+
+def test_rotate_z():
+    np_points = np.array([1, 1, 1])
+    pset = pyvista.PointSet(np_points)
+    pset.rotate_z(45, inplace=True)
+    assert np.array_equal(
```
Comment (akaszynski) [inline tests/test_pointset.py:251]:
```suggestion
    assert np.allclose(
```

File: tests/test_pointset.py [hunk 11]
```diff
@@ -106,3 +106,157 @@ def test_filters_return_pointset(sphere):
...
+        pset.points, np.array([[1.1102230e-16, 1.4142135e00, 1.0000000e00]], dtype=np.float32)
+    )
+
+
+def test_rotate_vector():
+    np_points = np.array([1, 1, 1])
+    pset = pyvista.PointSet(np_points)
+    pset.rotate_vector([1, 2, 1], 45, inplace=True)
+    assert np.array_equal(
```
Comment (akaszynski) [inline tests/test_pointset.py:260]:
```suggestion
    assert np.allclose(
```

File: tests/test_pointset.py [hunk 12]
```diff
@@ -106,3 +106,157 @@ def test_filters_return_pointset(sphere):
...
+            dtype=np.float32,
+        ),
+    )
+
+
+def test_flip_normal():
+    np_points = np.array([[1, 2, 3], [4, 5, 6], [7, 8, 9]])
+    pset = pyvista.PointSet(np_points)
+    pset.flip_normal([1.0, 1.0, 1.0], inplace=True)
+    assert np.array_equal(
```
Comment (akaszynski) [inline tests/test_pointset.py:212]:
```suggestion
    assert np.allclose(
```

File: tests/test_pointset.py [hunk 13]
```diff
@@ -106,3 +106,157 @@ def test_filters_return_pointset(sphere):
...
+            dtype=np.float32,
+        ),
+    )
+
+
+def test_flip_z():
+    np_points = np.array([[1, 2, 3], [4, 5, 6], [7, 8, 9]])
+    pset = pyvista.PointSet(np_points)
+    pset.flip_z(inplace=True)
+    assert np.array_equal(
```
Comment (akaszynski) [inline tests/test_pointset.py:191]:
```suggestion
    assert np.allclose(
```

File: tests/test_pointset.py [hunk 14]
```diff
@@ -106,3 +106,157 @@ def test_filters_return_pointset(sphere):
...
+            dtype=np.float32,
+        ),
+    )
+
+
+def test_flip_y():
+    np_points = np.array([[1, 2, 3], [4, 5, 6], [7, 8, 9]])
+    pset = pyvista.PointSet(np_points)
+    pset.flip_y(inplace=True)
+    assert np.array_equal(
```
Comment (akaszynski) [inline tests/test_pointset.py:170]:
```suggestion
    assert np.allclose(
```

File: tests/test_pointset.py [hunk 15]
```diff
@@ -106,3 +106,157 @@ def test_filters_return_pointset(sphere):
...
+    pset = pyvista.PointSet(np_points)
+    pset.scale(2, inplace=True)
+    assert np.array_equal(pset.points, np.array([[2.0, 4.0, 6.0]], dtype=np.float32))
+
+
+def test_flip_x():
+    np_points = np.array([[1, 2, 3], [4, 5, 6], [7, 8, 9]])
+    pset = pyvista.PointSet(np_points)
+    pset.flip_x(inplace=True)
+    assert np.array_equal(
```
Comment (akaszynski) [inline tests/test_pointset.py:149]:
```suggestion
    assert np.allclose(
```

File: tests/test_pointset.py [hunk 16]
```diff
@@ -106,3 +106,157 @@ def test_filters_return_pointset(sphere):
...
+    pset = pyvista.PointSet(np_points)
+    pset.translate((4, 3, 2), inplace=True)
+    assert np.array_equal(pset.center, np.array([5, 5, 5]))
+
+
+def test_scale():
+    np_points = np.array([1, 2, 3])
+    pset = pyvista.PointSet(np_points)
+    pset.scale(2, inplace=True)
+    assert np.array_equal(pset.points, np.array([[2.0, 4.0, 6.0]], dtype=np.float32))
```
Comment (akaszynski) [inline tests/test_pointset.py:142]:
```suggestion
    assert np.allclose(pset.points, [2.0, 4.0, 6.0])
```

File: tests/test_pointset.py [hunk 17]
```diff
@@ -106,3 +106,157 @@ def test_filters_return_pointset(sphere):
...
+    np_points = np.array([[1, 2, 3]], np.int64)
+    pset = pyvista.PointSet(np_points)
+    assert pset.points_to_double().points.dtype == np.double
+
+
+def test_translate():
+    np_points = np.array([1, 2, 3], np.int64)
+    pset = pyvista.PointSet(np_points)
+    pset.translate((4, 3, 2), inplace=True)
+    assert np.array_equal(pset.center, np.array([5, 5, 5]))
```
Comment (akaszynski) [inline tests/test_pointset.py:135]:
```suggestion
    assert np.allclose(pset.center, [5, 5, 5])
```

File: tests/test_pointset.py [hunk 18]
```diff
@@ -106,3 +106,157 @@ def test_filters_return_pointset(sphere):
     pointset = sphere.cast_to_pointset()
     clipped = pointset.clip()
     assert isinstance(clipped, pyvista.PointSet)
+
+
+@pytest.mark.parametrize("force_float,expected_data_type", [(False, np.int64), (True, np.float32)])
+def test_pointset_force_float(force_float, expected_data_type):
+    np_points = np.array([[1, 2, 3]], np.int64)
+    pset = pyvista.PointSet(np_points, force_float=force_float)
+
```
Comment (akaszynski) [inline tests/test_pointset.py:115]:
```suggestion
```

File: tests/test_pointset.py [hunk 19]
```diff
@@ -106,3 +106,140 @@ def test_filters_return_pointset(sphere):
...
+        ),
+    )
+
+
+def test_rotate_x():
+    np_points = np.array([1, 1, 1])
+    pset = pyvista.PointSet(np_points)
+    pset.rotate_x(45, inplace=True)
+    assert np.allclose(
+        pset.points, np.array([[1.0000000e00, 1.1102230e-16, 1.4142135e00]], dtype=np.float32)
```
Comment (akaszynski) [inline tests/test_pointset.py:217]:
```suggestion
        pset.points, [1.0000000e00, 1.1102230e-16, 1.4142135e00])
```

File: tests/test_pointset.py [hunk 20]
```diff
@@ -106,3 +106,140 @@ def test_filters_return_pointset(sphere):
...
+    )
+
+
+def test_rotate_y():
+    np_points = np.array([1, 1, 1])
+    pset = pyvista.PointSet(np_points)
+    pset.rotate_y(45, inplace=True)
+    assert np.allclose(
+        pset.points, np.array([[1.4142135e00, 1.0000000e00, 1.1102230e-16]], dtype=np.float32)
```
Comment (akaszynski) [inline tests/test_pointset.py:226]:
```suggestion
        pset.points, [1.4142135, 1.0, 0.0])
```

File: tests/test_pointset.py [hunk 21]
```diff
@@ -106,3 +106,140 @@ def test_filters_return_pointset(sphere):
...
+    )
+
+
+def test_rotate_x():
+    np_points = np.array([1, 1, 1])
+    pset = pyvista.PointSet(np_points)
+    pset.rotate_x(45, inplace=True)
+    assert np.allclose(
+        pset.points, np.array([[1.0000000e00, 1.1102230e-16, 1.4142135e00]], dtype=np.float32)
+    )
```
Comment (akaszynski) [inline tests/test_pointset.py:218]:
```suggestion
    assert np.allclose(pset.points, [1.0, 0.0, 1.4142135])
```

File: tests/test_pointset.py [hunk 22]
```diff
@@ -106,3 +106,140 @@ def test_filters_return_pointset(sphere):
...
+    pset.flip_normal([1.0, 1.0, 1.0], inplace=True)
+    assert np.allclose(
+        pset.points,
+        np.array(
+            [
+                [7.0, 8.0, 9.0],
+                [4.0, 5.0, 6.0],
+                [1.0, 2.0, 3.0],
+            ],
+            dtype=np.float32,
```
Comment (akaszynski) [inline tests/test_pointset.py:207]:
```suggestion
```

File: tests/test_pointset.py [hunk 23]
```diff
@@ -106,3 +106,140 @@ def test_filters_return_pointset(sphere):
...
+    )
+
+
+def test_rotate_y():
+    np_points = np.array([1, 1, 1])
+    pset = pyvista.PointSet(np_points)
+    pset.rotate_y(45, inplace=True)
+    assert np.allclose(
+        pset.points, np.array([[1.4142135e00, 1.0000000e00, 1.1102230e-16]], dtype=np.float32)
+    )
```
Comment (akaszynski) [inline tests/test_pointset.py:227]:
```suggestion
    assert np.allclose(pset.points, [1.4142135, 1.0, 0.0])
```

File: tests/test_pointset.py [hunk 24]
```diff
@@ -106,3 +106,140 @@ def test_filters_return_pointset(sphere):
...
+
+
+def test_rotate_vector():
+    np_points = np.array([1, 1, 1])
+    pset = pyvista.PointSet(np_points)
+    pset.rotate_vector([1, 2, 1], 45, inplace=True)
+    assert np.allclose(
+        pset.points, np.array([[1.1910441, 1.0976311, 0.6136938]], dtype=np.float32)
+    )
```
Comment (akaszynski) [inline tests/test_pointset.py:245]:
```suggestion
    assert np.allclose(pset.points, [1.1910441, 1.0976311, 0.6136938])
```

File: tests/test_pointset.py [hunk 25]
```diff
@@ -106,3 +106,140 @@ def test_filters_return_pointset(sphere):
...
+    pset.flip_z(inplace=True)
+    assert np.allclose(
+        pset.points,
+        np.array(
+            [
+                [1.0, 2.0, 9.0],
+                [4.0, 5.0, 6.0],
+                [7.0, 8.0, 3.0],
+            ],
+            dtype=np.float32,
```
Comment (akaszynski) [inline tests/test_pointset.py:190]:
```suggestion
```

File: tests/test_pointset.py [hunk 26]
```diff
@@ -106,3 +106,140 @@ def test_filters_return_pointset(sphere):
...
+    pset.flip_y(inplace=True)
+    assert np.allclose(
+        pset.points,
+        np.array(
+            [
+                [1.0, 8.0, 3.0],
+                [4.0, 5.0, 6.0],
+                [7.0, 2.0, 9.0],
+            ],
+            dtype=np.float32,
```
Comment (akaszynski) [inline tests/test_pointset.py:173]:
```suggestion
```

File: tests/test_pointset.py [hunk 27]
```diff
@@ -106,3 +106,140 @@ def test_filters_return_pointset(sphere):
...
+    pset.flip_x(inplace=True)
+    assert np.allclose(
+        pset.points,
+        np.array(
+            [
+                [7.0, 2.0, 3.0],
+                [4.0, 5.0, 6.0],
+                [1.0, 8.0, 9.0],
+            ],
+            dtype=np.float32,
```
Comment (akaszynski) [inline tests/test_pointset.py:156]:
```suggestion
```

**Review by akaszynski — APPROVED** [review]
Made some additional style changes in line with @adeak and @tkoyama010's comments.

LGTM. Thanks!

**Review by tkoyama010 — APPROVED** [review]
LGTM

**Pull request merged**